src/java/org/apache/cassandra/index/sai/plan/QueryController.java

```diff
@@ -295,16 +300,24 @@ public static IndexTarget.Type determineIndexTargetType(RowFilter.Expression exp
     /**
      * Get an iterator over the rows for this partition key. Builds a search view that includes all memtables and all
      * {@link SSTableSet#LIVE} sstables.
-     * @param key
+     * @param keys
      * @param executionController
      * @return
      */
-    public UnfilteredRowIterator getPartition(PrimaryKey key, ReadExecutionController executionController)
+    public UnfilteredRowIterator getPartition(List<PrimaryKey> keys, ReadExecutionController executionController)
     {
-        if (key == null)
+        if (keys == null)
             throw new IllegalArgumentException("non-null key required");
-
-        SinglePartitionReadCommand partition = getPartitionReadCommand(key, executionController);
+        if (keys.isEmpty())
+            throw new IllegalArgumentException("At least one primary key is required!");
+
+        SinglePartitionReadCommand partition = SinglePartitionReadCommand.create(cfs.metadata(),
```

Comment:
As an alternative approach, would it make any sense to cache all the partition's clustering keys and ask for the entire partition if there are more clusterings than certain threshold, to mimic `aa`?

Reply:
I think that it is better to stick to the original patch for now, then we can iterate